### PR TITLE
Addon Docs! :tada:

### PR DESCRIPTION
**Pull request by lolmaus**

### CODE REVIEW

File: tests/dummy/app/pods/docs/truth/template.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+
+
+## Pros and cons of this approach
+
+Let's start with **cons**.
+
+The suggested approach:
+
+* Makes feature files more explicit and more techinical.
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/truth/template.md]:
```suggestion
* Makes feature files more explicit and more technical.
```

File: tests/dummy/app/pods/docs/truth/template.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    * Easy to navigate thanks to documentation available here.
+    * Little to no maintenance.
+    * You no longer need to have a steps file for each acceptance test. Why bother, when all steps are fully reusable across all acceptance tests?
+    * The amount of regexp collisions between steps is reduced drastically.
+* As you're making new tests:
+    * You don't need to code step implementations.
+    * Writing new tests is fast and enjoyable.
+    * You can expand existing features with more cases by simply copying the first case and adjusting the options. Not that you weren't able to do this before, but now it's even easier.
+* The truth is no longer hidden inside the black box of step implementations:
+    * It is now possible to reliable validate features: by reading them, ensure they are making correct assertions in correct connditions.
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/truth/template.md]:
```suggestion
    * It is now possible to reliable validate features: by reading them, ensure they are making correct assertions in correct conditions.
```

File: tests/dummy/app/pods/docs/labels/template.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+As explained <LinkTo @route="docs.truth">previously</LinkTo>, we want to move the truth from step implementations to feature files. A great way to do it is with test selectors.
+
+To learn about test selectors, please see the [ember-test-selectors](https://github.com/simplabs/ember-test-selectors) addon. Though you can start using test selectors without the addon, it is recommended to have it installed in your app.
+
+`ember-cli-yadda-opinonated` introduces *labels*: a DSL for your Cucumber feature files that lets you reference page elements via test selectors.
+
+In its simplest form, a label can look like: `Menu-Item`, `MenuItem` or `menu-item`. Any of these lables translates to `[data-test-menu-item]`.
+
+You can also use article. They will be ignored by the addon, but will contribute to readability: `a Menu-Item` and `the Menu-Item` behave identically to `Menu-Item`.
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/labels/template.md]:
```suggestion
You can also use articles. They will be ignored by the addon, but will contribute to readability: `a Menu-Item` and `the Menu-Item` behave identically to `Menu-Item`.
```

File: tests/dummy/app/pods/docs/about/template.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+#### BDD, in a narrow sense
+
+An approach to coding which adds a second loop to the TDD cycle: you are supposed to write an acceptance test before writing a unit test. Also, a different style of writing assertions: `expect` or `should` instead of `assert`.
+
+
+
+#### BDD, in a broad sense
+
+
+A methodology of software devleopment, which involves all the team, not only developers. When a feature is conceieved, it is specced out in form of human-readable user stories. These user stories serve as a spec and are used to guide and validate development from planning to deployment.
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/about/template.md]:
```suggestion
A methodology of software development, which involves all the team, not only developers. When a feature is conceived, it is specced out in form of human-readable user stories. These user stories serve as a spec and are used to guide and validate development from planning to deployment.
```

File: tests/dummy/app/pods/docs/about/template.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# What this addon is
+
+Let's start with some glossary.
+
+
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/about/template.md]: I thinks something like this has to be part of the introduction! When looking at the `Why cucumber` page I wondered how anyone who does not know what Cucumber or BDD is should understand it. I think that would help!

File: tests/dummy/app/pods/docs/template.hbs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    {{nav.item "Using step aliases" "docs.step-aliases"}}
+    {{nav.item "Mapping labels to selectors" "docs.label-map"}}
+    {{nav.item "Making assertions" "docs.assertions"}}
+
+    {{nav.section "Steps library"}}
+    {{nav.item "Given steps" "docs.steps-given"}}
+    {{nav.item "When steps" "docs.steps-when"}}
+    {{nav.item "Then steps" "docs.steps-then"}}
+    {{nav.item "Power Select steps" "docs.steps-power-select"}}
+    {{nav.item "Power Date Picekr steps" "docs.steps-power-date-picker"}}
```
Comment (simonihmig) [inline tests/dummy/app/pods/docs/template.hbs]:
```suggestion
    {{nav.item "Power Date Picker steps" "docs.steps-power-date-picker"}}
```

File: tests/dummy/app/pods/index/index-content/template.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+## Why Cucumber
+
+Cucumber is a famous BDD implementation that gives exciting promises.
+
+
+
+#### Acceptance tests in form of <em>user stories</em> offer a lot of benefits:
```
Comment (simonihmig) [inline tests/dummy/app/pods/index/index-content/template.md]: Idk why, but this renders as a link, which points to itself. Which is, weird.

Also noticed that these links are rendered with the default blue of the browser, not the blue defined in the addon-docs theme as all the other "real" links...

**Review by simonihmig — COMMENTED** [review]
Did only review the introduction and usage chapters. Apart from some typos I would recommend to work on the introduction for BDD newbies, see the other comment. Other than that, this looks awesome! 🚀

File: tests/dummy/app/pods/index/template.hbs [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<DocsHeader />
+
+<DocsHero
+  @heading = "Yadda Opinionated"
+  @byline  = "A radical take on Cucumber to solve you test maintenance woes"
```
Comment (simonihmig) [inline tests/dummy/app/pods/index/template.hbs]: Is that correct? Or should it be "your"?

File: tests/dummy/app/pods/docs/about/template.md (hunk 5 above)
Comment (lolmaus) [reply tests/dummy/app/pods/docs/about/template.md]: Done.

File: tests/dummy/app/pods/index/index-content/template.md (hunk 7 above)
Comment (nickschot) [reply tests/dummy/app/pods/index/index-content/template.md]: Maybe 4 # does not exist and it sees the leftover # as a hash url?

Comment (lolmaus) [reply tests/dummy/app/pods/index/index-content/template.md]: > Idk why, but this renders as a link, which points to itself. Which is, weird.

Headers are intentionally links anchored to themselves! This lets you copy a link to a specific section of the page, not to the top of the page.

Note that it works with pure HTML, no JS involved.

***

> Also noticed that these links are rendered with the default blue of the browser

This is an upstream issue. I have worked around it by using higher-ranked h* tags which are not affected.

File: tests/dummy/app/pods/index/template.hbs (hunk 8 above)
Comment (lolmaus) [reply tests/dummy/app/pods/index/template.hbs]: Fixed.

**Comment by lolmaus** [comment]
> Apart from some typos I would recommend to work on the introduction for BDD newbies, see the other comment.

I did as you suggested in the other comment, but I don't want to give a bigger introduction to BDD than what we already have.

Teaching people BDD is just out of scope of this addon.

Also, if you have reached this addon's docs, then there's almost a 100% chance that you are already be doing BDD with Cucumber, or at least are interested in it enough to discover such a specialized thing as this addon.

**Pull request merged**